enhancements/cluster-logging/cluster-logging-es-rollover-data-design.md

```diff
@@ -0,0 +1,246 @@
+* Security will be addressed by using the OpenDistro security plugin and document level security (DLS). Details TBD.
+
+#### Data Model
+Logs of a given type are co-located to the following indices:
+
+| Log Type | Read Alias | Write Alias | Initial Index |
+|---------|-----------|---------------|---------------|
+| Node (`logs.infra`)|infra, infra.node|infra.node-write|node.infra-00001|
+| Container Infra (`logs.infra`)|infra, infra.container|infra.container-write|container.infra-000001|
+| Container Application (`logs.app`)|app.logs|app.container-write|container.app-000001|
+| Audit (`logs.audit`)|infra.audit|infra.audit-write|audit.infra-000001|
+
+**Note:** Log types are further defined in [LogForwarding](./cluster-logging-log-forwarding.md).
+
+#### ClusterLogging API
+```
+apiVersion: "logging.openshift.io/v1"
+kind: "ClusterLogging"
+metadata:
+  name: "instance"
+spec:
+  logStore:
+    retentionPolicy: 
+      logs.app:
+        maxAge: 7d
+      logs.infra
+        maxAge: 7d
+
+```
+#### Cluster Logging Operator
+The `cluster-logging-operator` will utilize the ClusterLogging CR retention policy to spec
+the desired aforementioned Elasticsearch CR indexManagement.
+#### Elasticsearch API
+```
+apiVersion: "logging.openshift.io/v1"
+kind: "Elasticsearch"
+metadata:
+  name: "elasticsearch"
+spec:
+  indexManagement:
+    policies:
+    - name: infra-policy
+      pollInterval: 5m
+      phases:
+        hot:
+          actions:
+            rollover:
+              maxAge:   3d
+        delete:
+          minAge: 7d
+    mappings:
+    - name:  node.infra               #creates node.infra-00001 aliased node.infra-write
+      policyRef: infra-policy         #policy applies to index patterns node.infra*
+      aliases:
+      - infra  
+```
+#### Elasticsearch Operator
+The `elasticsearch-operator` will be modified to:
+* Expose the index management API
+* Create and seed index templates to support the policy
+* Create the initial indices as needed
+* Block data ingestion if needed until initial index is seeded
+* Deploy curation CronJob for each mapping to rollover using the defined policy
+
+#### Collector 
+* Update the [`fluent-plugin-viaq-data-model`](https://github.com/ViaQ/fluent-plugin-viaq_data_model) to allow defining a static index to write logs
+
+Example configuration:
+```    
+  ....
+   <elasticsearch_index_name>
+     tag "**"
+     name_type static index_name=>'container.app-write'
+   </elasticsearch_index_name>
+```
+#### Deprecations
+* The Curator Cronjob deployed by the `cluster-logging-operator` will be removed. The responsibilities for curation will be subsumed by
+ implementation of Elasticsearch rollover management.
+* Curation configuration by namespace is no longer configurable and is restricted to cluster
+wide settings associated with log type 
```

Comment:
I am not a fan of not having a possibility to control amount of logs per project. It is fair to say that it comes with risks.

Say some developer turns on `TRACE` level or makes unwise decision (or simply a mistake) and one or a few projects start generating a lot of logs. As of now it is at least possible to go and ad-hoc delete individual history indices for project. But when all logs are blend into a common index there is no such possibility and all other projects will get hit as well (even whole ES cluster can easily suffer) with no simple solution (except from curating all the common logs for projects sharing this index).

If we can not control this on a project level then we should at least make sure we ship appropriate alert rules to inform cluster admins that logs suddenly started growing much faster and something needs to be done about this.

Reply:
This is the reason why we need to node-level controls for logging rates, see https://github.com/containers/conmon/issues/84

Reply:
@lukas-vlcek This change only would deprecate namespace curation for now.  We have discussed modifying LogForwarding to allow include/exclude which maybe its possible to also address throttling.